Guidance on selecting a persister (#4280)
Particular/docs.particular.net: persistence/index.md

---
title: Persistence
summary: Features of NServiceBus requiring persistence include timeouts, sagas, and subscription storage.
reviewed: 2019-02-27
redirects:
- nservicebus/persistence-in-nservicebus
- nservicebus/persistence
---

> NServiceBus needs to store data for various purposes, like persisting the state of sagas, enable the outbox and on some occasions to make a transport feature complete.

Reviewer: ```suggestion
NServiceBus needs to store data for various purposes, such as persisting the state of sagas, enabling the outbox, and for certain other transport features.
```

## Features that require persistence

 * [Sagas](/nservicebus/sagas/)
 * [Outbox](/nservicebus/outbox/)
 * [Subscriptions](/nservicebus/messaging/publish-subscribe/)
 * [Timeouts](/nservicebus/sagas/timeouts.md)
 * [Delayed Retries](/nservicebus/recoverability/#delayed-retries)
 * [Deferral](/nservicebus/messaging/delayed-delivery.md)
 * [Gateway Deduplication](/nservicebus/gateway/)

> ## How to select a persister

Reviewer: ```suggestion
## Selecting a persister
```
